src/components/ListingItems/ListingItems.jsx

```diff
@@ -0,0 +1,266 @@
+import { addDoc, collection } from "firebase/firestore"
+import Image from "next/image"
+import { useTranslation, withTranslation } from "next-i18next"
+import { useState } from "react"
+
+import { db } from "@/utils/firebase/config"
+
+import Button from "../button"
+import Highlighter from "../highlighter"
+import Input from "../input"
+
+function ListingItems() {
+  const { t } = useTranslation("listingItems")
+  // Form data handler
+  const [formData, setFormData] = useState({
+    primaryImage: { file: null, url: "/images/emptyImage.png" },
+    secondaryImage: { file: null, url: "/images/emptyImage.png" },
+    tertiaryImage: { file: null, url: "/images/emptyImage.png" },
+    quaternaryImage: { file: null, url: "/images/emptyImage.png" },
+    type: "",
+    category: "",
+    productName: "",
+    description: "",
+    location: "",
+    price: "",
+  })
+
+  // Input handler
+  const inputsHandler = (event) => {
+    const { name, value } = event.target
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }))
+  }
+
+  // Image upload handler
+  const uploadedImageHandler = (event, imageField) => {
```

Comment:
i tried this way which is not working with the firebase because of that the firestore never collect a files when i tried this way it gives me an error which say you are trying to collect a file to the database this way is wrong so that the fles can only had been uploaded to the storage only